.github/workflows/codeql.analysis.yml

```diff
@@ -0,0 +1,55 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '32 11 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # CodeQL also supports Java, but our build system is too esoteric for them.
+        language: [ 'go', 'javascript', 'python' ]
```

Comment:
This GHA is set to run for commits to master and on PRs. As-is, all 3 languages are scanned each time. What do you think about breaking this workflow file into 3 separate workflow files so each language's workflow is only triggered if sources for that language are changed?

Reply:
So I think the reason it isn't is because when you run the cron task you want it to hit everything, since nothing in particular has changed, possibly. I could split it into 4 separate worflows to keep that behavior and also only run what's needed on PRs.

Reply:
So I think the reason it isn't is because when you run the cron task you want it to hit everything, since nothing in particular has changed, possibly. I could split it into 4 separate worflows to keep that behavior and also only run what's needed on PRs.

Reply:
Yeah, with that explanation, 4 workflows makes the most sense since we still want the weekly job to hit all languages.